prime-router/src/main/kotlin/fhirengine/engine/FHIRRouter.kt

```diff
@@ -41,40 +41,94 @@ class FHIRRouter(
-        "Bundle.entry.resource.ofType(MessageHeader).id.exists()",
-        "Bundle.entry.resource.ofType(Patient).name.family.exists()",
-        "Bundle.entry.resource.ofType(Patient).name.given.count() > 0",
-        "Bundle.entry.resource.ofType(Patient).birthDate.exists()",
-        "Bundle.entry.resource.ofType(Specimen).type.exists()",
-        "(Bundle.entry.resource.ofType(Patient).address.line.exists() or " +
-            "Bundle.entry.resource.ofType(Patient).address.postalCode.exists() or " +
-            "Bundle.entry.resource.ofType(Patient).telecom.exists())",
+        "%messageId.exists()",
+        "%patient.name.family.exists()",
+        "%patient.name.given.count() > 0",
+        "%patient.birthDate.exists()",
+        "%specimen.type.exists()",
+        "(%patient.address.line.exists() or " +
+            "%patient.address.postalCode.exists() or " +
+            "%patient.telecom.exists())",
         "(" +
-            "(Bundle.entry.resource.ofType(Specimen).collection.collectedPeriod.exists() or " +
-            "Bundle.entry.resource.ofType(Specimen).collection.collected.exists()" +
+            "(%specimen.collection.collectedPeriod.exists() or " +
+            "%specimen.collection.collected.exists()" +
             ") or " +
-            "Bundle.entry.resource.ofType(ServiceRequest).occurrence.exists() or " +
-            "Bundle.entry.resource.ofType(Observation).effective.exists())"
+            "%serviceRequest.occurrence.exists() or " +
+            "%observation.effective.exists())"
     )
 
     /**
      * Default Rule:
      *  Must have a processing mode id of 'P'
      */
     val processingModeFilterDefault: ReportStreamFilter = listOf(
-        "Bundle.entry.resource.ofType(MessageHeader).meta" +
-            ".extension('https://reportstream.cdc.gov/fhir/StructureDefinition/source-processing-id')" +
-            ".value.coding.code = 'P'"
+        "%processingId = 'P'"
     )
 
+    /**
+     * Lookup table `fhirpath_filter_shorthand` containing all of the shorthand fhirpath replacements for filtering.
+     */
+    private val shorthandLookupTable by lazy { loadFhirPathShorthandLookupTable() }
+
+    /**
+     * Constants to make writing filter conditions shorter / more accessible. This will replace the shorthand
+     * used in configuration filter expressions with the specified Fhir Path expression before the expression
+     * is evaluated against the bundle. This allows for returning of collections, as well as handling 'exists()'
+     * where needed. Format is %myVariable within the filters, and each variable used must be defined
+     * as part of the shorthand collection of an exception will be raised. Make sure to verify if you need to use
+     * 'exists' on your check - an 'unable to parse' will return 'false' but will not raise any other error due to the
+     * underlying fhir library.
+     *
+     * The values used are located in the fhirpath_filter_shorthand lookup table.
+     */
```

Comment:
I think this is the kdoc for the function below. Missing @ return.  The regex needs kdoc and would be nice if you write about what it covers